presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -247,97 +240,56 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
     {
         InformationSchemaTableHandle table = (InformationSchemaTableHandle) handle;
 
-        if (!table.getPrefixes().equals(defaultPrefixes())) {
+        // The table is catalog-wise and doesn't have a schema/table name filter.
+        if (!isTablesEnumeratingTable(table.getSchemaTableName())) {
             return Optional.empty();
         }
 
-        Set<QualifiedTablePrefix> prefixes = getPrefixes(session, table, constraint);
-
-        table = new InformationSchemaTableHandle(table.getCatalogName(), table.getSchemaName(), table.getTableName(), prefixes);
-
-        return Optional.of(new ConstraintApplicationResult<>(table, constraint.getSummary()));
-    }
-
-    private Set<QualifiedTablePrefix> defaultPrefixes()
-    {
-        return ImmutableSet.of(new QualifiedTablePrefix(catalogName));
-    }
+        // TODO: constraint may have two levels of filters: schema-level and table-level. It also has two types of filters: discrete (=) and range (>, <, etc.).
+        //   Discrete and range schema-level filters are supported;
+        //   discrete tabel-level filters are supported;
+        //   range table-level filters are not yet supported.
+        Optional<Set<String>> newSchemaNames = filterString(constraint.getSummary(), SCHEMA_COLUMN_HANDLE);
+        if (!newSchemaNames.isPresent()) {
+            if (table.getSchemaNames().isPresent()) {
+                newSchemaNames = table.getSchemaNames();
```

Comment:
`table.getSchemaNames();` is more like old schema names. I don't get it why it is assigned to `newSchemaNames`.